runtime/Java/src/org/antlr/v4/runtime/tree/IterativeParseTreeWalker.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * [The "BSD license"]
+ *  Copyright (c) 2012 Terence Parr
```

Comment:
The year is particularly interesting. It turns out keeping the year constant and using the year when the code base started provides the longest and strongest copyright protections. If you want to see some of the details on this, take a look at the long and probably very boring comments in DotNetAnalyzers/StyleCopAnalyzers#1661.

The use of mine and Terence's names has to do with copyright assignment, which is (to the best of my knowledge) common practice in a number of circles. There are three ways we work to ensure the original author of patches remains recognized for the life of the project:

1. The **contributors.txt** file, which is an essential and core part of this project. It's not distributed, but the distribution links back to it.
2. The Git commit history.
3. The pull requests and issues filed by everyone.

:memo: I am not a lawyer